libs/clients/health-directorate/src/lib/healthDirectorateClient.service.ts

```diff
@@ -137,6 +150,57 @@ export class HealthDirectorateClientService {
     return result
   }
 
+  async getHealthCareLicensesForWorkPermit(
+    auth: Auth,
+  ): Promise<StarfsleyfiUmsoknStarfsleyfi[]> {
+    const licenses = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiStarfsleyfiGet()
+
+    return licenses
+  }
+
+  async getHealthCareWorkPermitEducationInfo(
+    auth: Auth,
+  ): Promise<NamsUpplysingar[]> {
+    const educationInfo = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiNamsUpplysGet()
+
+    return educationInfo
+  }
+
+  async submitApplicationHealthcareWorkPermit(
+    auth: User,
+    request: HealthcareWorkPermitRequest,
+  ): Promise<UtbuaStarfsleyfiSkjalResponse[]> {
+    const items = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiUtbuaSkjalPost({
+      utbuaStarfsleyfiSkjalRequest: {
+        name: request.name,
+        dateOfBirth: format(new Date(request.dateOfBirth), 'dd.MM.yyyy'),
+        citizenship: request.citizenship,
+        email: request.email,
+        phoneNo: request.phone,
+        idProfession: request.idProfession,
+        education: request.education,
+      },
+    })
+
+    if (!items || !Array.isArray(items) || items.length === 0) {
+      throw new Error('No items returned')
+    }
+
+    for (const item of items) {
+      if (!item.base64String) {
+        throw new Error('Empty file in one of the items')
```

Comment:
If this is an actual error, then we must log it, and have a more descriptive error message! I.e. who's causing the issue, why it's an error, etc. The more detailed the logging, the better.

Reply:
Logging added.

Reply:
Sorry, after some discussion it seems that logging and throwing is a bit much. Either log the error here and return null, or throw , catch on a higher level and log there, or just let it bubble for the stack trace.